rpc/json/ws.go

```diff
@@ -33,13 +33,40 @@ func (wsc *wsConn) sendLoop() {
 	}
 }
 
+func CheckOrigin(r *http.Request, allowedOrigins []string) bool {
```

Comment:
Couldn't we reuse some existing, well tested function from some CORS library?
This is security-related function (prevents XSS) - I think we should reuse something.
Also, I'm not 100% sure about expected semantics of this function (see comments below).

Absolute minimum - cover this with tests.

Reply:
I don't think there is any XSS vulnerability here since there is no requirement for a browser origin for the websocket rpc client. In fact, I think we can take the same approach as tendermint and accept _all_ origins, headers and methods without any security issue as such:

https://github.com/tendermint/tendermint/blob/35581cf54ec436b8c37fabb43fdaa3f48339a170/rpc/jsonrpc/server/ws_handler.go#L49

Reply:
Yeah I thought about this but took a middle way between tendermint's _very_ permissive approach and a strict verification approach

https://github.com/tendermint/tendermint/blob/35581cf54ec436b8c37fabb43fdaa3f48339a170/rpc/jsonrpc/server/ws_handler.go#L49

I'm not sure there is a XSS vulnerability here as we are not expecting the RPC server to be restricted to a particular origin.

However the tendermint approach is misleading as it seems to accept the config variables only to never use them to restrict the origin.

Not sure what the right way to deal with this is, but happy to discuss ideas.